src/gameComponents/__tests__/Areas.spec.ts

```diff
@@ -0,0 +1,100 @@
+import { shallowMount, Wrapper } from '@vue/test-utils'
+import Areas from '@/gameComponents/Areas.vue'
+import Vue from 'vue'
+
+const baseMocks = {
+  $store: {
+    state: {
+      scoreStore: {
+        score: 0,
+        lastStyle: 'excellent'
+      },
+      arrowStore: {
+        arrowsData:
+          [{
+            id: 1,
+            direction: 'left-arrow'
+          }]
+      }
+    }
+  }
+}
+
+let wrapper: Wrapper<Areas>
+let vm: Areas | any
+
+describe('Areas.vue', () => {
+  beforeEach(() => {
+    wrapper = shallowMount(Areas, {
+      mocks: baseMocks
+    })
+    vm = wrapper.vm
+  })
+
+  it('animationStyle must consist necessary style', () => {
+    expect(vm.animationStyle).toEqual({
+      good: '',
+      excellent: 'excellent-animation'
+    })
+  })
+
+  it('emit should exist and return necessary obj', async () => {
+    vm.$emit('calculate-positions')
+    await Vue.nextTick()
+    expect(wrapper.emitted('calculate-positions')).toHaveLength(2)
+    expect(wrapper.emitted('calculate-positions')).toMatchObject([
+      [{
+        bottomExcellentArea: 0,
+        bottomGoodArea: 0,
+        topExcellentArea: 0,
+        topGoodArea: 0
+      }],
+      []
+    ])
```

Comment:
Nice
Could you explain what this test checks? Do you test that you called  `calculate-positions` event once, but it was called twice? Once it returned  
```
{
        bottomExcellentArea: 0,
        bottomGoodArea: 0,
        topExcellentArea: 0,
        topGoodArea: 0
}
```
and the second time it returned nothing? Once it was called in `mounted` and the second time - `vm.$emit('calculate-positions')`? What differs these two calls?

Reply:
or I need to add `expect(wrapper.emitted('calculate-positions')).toHaveLength(1)`?